common/test/acceptance/tests/test_cohorted_courseware.py

```diff
@@ -0,0 +1,226 @@
+        """
+        View content as staff, student in Cohort A, student in Cohort B, and student in Default Cohort.
+        """
+        courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        def login_and_verify_visible_problems(username, email, expected_problems):
+            LmsAutoAuthPage(
+                self.browser, username=username, email=email, course_id=self.course_id
+            ).visit()
+            courseware_page.visit()
+            self.assertTrue(courseware_page.xblock_components_contain_strings(expected_problems))
+
+        login_and_verify_visible_problems(
+            self.staff_user["username"], self.staff_user["email"],
+            [self.group_a_problem, self.group_b_problem, self.group_a_and_b_problem, self.visible_to_all_problem]
+        )
+
+        login_and_verify_visible_problems(
+            self.cohort_a_student_username, self.cohort_a_student_email,
+            [self.group_a_problem, self.group_a_and_b_problem, self.visible_to_all_problem]
+        )
+
+        login_and_verify_visible_problems(
+            self.cohort_b_student_username, self.cohort_b_student_email,
+            [self.group_b_problem, self.group_a_and_b_problem, self.visible_to_all_problem]
+        )
+
+        login_and_verify_visible_problems(
+            self.cohort_default_student_username, self.cohort_default_student_email,
+            [self.visible_to_all_problem]
+        )
+
+    def test_cohorted_courseware(self):
+        """
+        Scenario: Can create content that is only visible to students in particular cohorts
+          Given that I have course with 4 problems, 1 staff member, and 3 students
+          When I enable cohorts in the course
+          And I create two content groups, Content Group A, and Content Group B, in the course
+          And I link one problem to Content Group A
+          And I link one problem to Content Group B
+          And I link one problem to both Content Group A and Content Group B
+          And one problem remains unlinked to any Content Group
+          And I create two manual cohorts, Cohort A and Cohort B,
+            linked to Content Group A and Content Group B, respectively
+          And I assign one student to each manual cohort
+          And one student remains in the default cohort
+          Then the staff member can see all 4 problems
+          And the student in Cohort A can see all the problems except the one linked to Content Group B
+          And the student in Cohort B can see all the problems except the one linked to Content Group A
+          And the student in the default cohort can ony see the problem that is unlinked to any Content Group
+        """
+        self._enable_cohorts_in_course()
+        self._create_content_groups()
+        self._link_problems_to_content_groups_and_publish()
+        self._create_cohorts_and_assign_students()
+        self._view_cohorted_content_as_different_users()
```

Comment:
Also: I know that this test is doing a lot, but it might be more readable if the helper methods were to accept semantic parmeters. For example, _create_content_groups() could take a list of content groups to add. Then again, I realize the helpers are each being used only once, so they can afford not taking arguments. And the docstring describes the logic. Nevermind.